zebra-state/src/service.rs

```diff
@@ -64,21 +66,40 @@ impl StateService {
 
     pub fn new(config: Config, network: Network) -> Self {
         let disk = FinalizedState::new(&config, network);
+
         let mem = NonFinalizedState {
             network,
             ..Default::default()
         };
         let queued_blocks = QueuedBlocks::default();
         let pending_utxos = PendingUtxos::default();
 
-        Self {
+        let state = Self {
             disk,
             mem,
             queued_blocks,
             pending_utxos,
             network,
             last_prune: Instant::now(),
+        };
+
+        tracing::info!("starting legacy chain check");
+        if let Some(tip) = state.best_tip() {
+            if legacy_chain_check(tip.0, state.any_ancestor_blocks(tip.1), state.network).is_err() {
```

Comment:
Bug: `tip.0` is not the NU5 activation height.

Reply:
Yea, you are right here but what can we send, `Canopy` with a `TODO` to change after we have `Nu5` height?

Reply:
We can skip the check if there is no NU5 activation height, because the next possible legacy chain is Canopy -> NU5.

That way, we automatically do the check on testnet as soon as its height is added, but the mainnet check waits for a few months.

```rust
if let Some(tip) = state.best_tip() {
    if let Some(nu5_activation_height) = Nu5.activation_height(network) {
```

Reply:
Thanks. I added this at https://github.com/ZcashFoundation/zebra/pull/2366/commits/922d36605113d2de7fb93fdffe306891d6dffb42